lib/boxxspring/worker/base.rb

```diff
@@ -58,31 +61,55 @@ def process
         messages.each do | message |
           if message.present?
             payload = self.payload_from_message( message )
+
             if payload.present?
               begin
                 result = self.process_payload( payload )
                 # note: if an exception is raised the message will be deleted
                 self.delete_message( message ) unless result == false
               rescue StandardError => error
                 self.logger.error(
-                  "The #{ self.human_name } worker failed to process the payload."
+                  "The #{ self.human_name } failed to process the payload."
                 )
                 self.logger.error( error.message )
-                self.logger.error( error.backtrace.join( "\n" ) )
+                self.logger.info( error.backtrace.join( "\n" ) ) if debug_mode?
               end
             else
-              # note: messages with invalid payloads are deleted
               self.delete_message( message )
               self.logger.error(
-                "The #{ self.human_name } worker received an invalid payload."
+                "The #{ self.human_name } received an invalid payload."
               )
             end
           end
         end
       end
 
-      protected; def logger
-        @logger ||= Boxxspring::Worker.configuration.logger
+      def logger
+        @logger ||= begin
+          worker_name = human_name.gsub( ' ','_' )
+          worker_env = self.class.environment
+          port = ( worker_env == 'production' ) ? 54323 : 48686
+          host_suffix = ( worker_env == 'production' ) ? '' : worker_env
+
+          Boxxspring::Worker.configuration do
+            logger(
+              RemoteSyslogLogger.new(
+                'logs2.papertrailapp.com',
+                port,
+                program: worker_name,
```

Comment:
Initialize the logger here (instead of initialization) so it can use the specific worker name as the remote logger program name.